src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -272,11 +272,6 @@ public final void beginTree(DetailAST rootAST) {
         TREE_CACHE.get().clear();
     }
 
-    @Override
-    public final void finishTree(DetailAST rootAST) {
```

Comment:
Removing this code allows anyone to override `finishTree` which is a java method. Since this is for Javadoc checks, it shouldn't have access to this type of method without some type of issue and reasoning.

Reply:
I missed `final` modifier. But anyway it looks a bit weird, because:
1. Comment says it no code _by default_, but it is not a default, it is the only option.
2. Usually some exception is thrown to indicate that this method is not used here, e.g. UnsupportedOperationException("This method shouldn't be used in javadoc check")

Reply:
> Usually some exception is thrown to indicate that this method is not used here

The method is still used by TreeWalker. Throwing an exception would break the check. It is empty and final so another Javadoc check can't override it and change it, but we don't need anything special from it.

> Comment says it no code by default, but it is not a default, it is the only option.

Changing the comment is fine if it is confusing.

Reply:
Are you ok with `No code, just to avoid override in heirs`?

Reply:
How about `no code, prevent override in subclasses`

Reply:
Sounds better, updated

Reply:
Thanks.